manifests/jenkins-s2i.yaml

```diff
@@ -0,0 +1,109 @@
+apiVersion: v1
+kind: Template
+labels:
+  app: fedora-coreos
+  template: fedora-coreos-jenkins-s2i-template
+metadata:
+  annotations:
+    description: |-
+      Jenkins S2I builder for Fedora CoreOS.
+    iconClass: icon-jenkins
+    openshift.io/display-name: Fedora CoreOS Jenkins S2I
+    openshift.io/documentation-url: https://github.com/coreos/fedora-coreos-pipeline
+    openshift.io/support-url: https://github.com/coreos/fedora-coreos-pipeline
+    openshift.io/provider-display-name: Fedora CoreOS
+    tags: fcos,jenkins,fedora
+  name: fedora-coreos-jenkins-s2i
+parameters:
+  - description: Git source URI for Jenkins S2I
+    name: JENKINS_S2I_URL
+    value: https://github.com/coreos/fedora-coreos-pipeline
+  - description: Git branch/tag reference for Jenkins S2I
+    name: JENKINS_S2I_REF
+    value: master
+  - description: Git source URI for Jenkins jobs
+    name: JENKINS_JOBS_URL
+    value: https://github.com/coreos/fedora-coreos-pipeline
+  - description: Git branch/tag reference for Jenkins jobs
+    name: JENKINS_JOBS_REF
+    value: master
+
+objects:
+
+  ### JENKINS MASTER ###
+
+  # use our own "gated" imagestream for the Jenkins master so we can test new
+  # images before upgrading
+  - apiVersion: v1
+    kind: ImageStream
+    metadata:
+      name: jenkins-s2i
+      annotations:
+        coreos.com/deploy-default: "true"
+    spec:
+      tags:
+        - name: stable
+          from:
+            kind: DockerImage
+            # 2.204.1
+            name: docker.io/openshift/jenkins-2-centos7@sha256:9b75175c6feabfac03f39af5342b31cb4931b53ad9d077d843bb2f1e2d9eb408
+  - apiVersion: v1
+    kind: ImageStream
+    metadata:
+      name: jenkins
+      annotations:
+        coreos.com/deploy-default: "true"
+  - kind: BuildConfig
+    apiVersion: v1
+    metadata:
+      name: jenkins
+      annotations:
+        coreos.com/deploy-default: "true"
+        coreos.com/jenkins-jobs-url: ${JENKINS_JOBS_URL}
+        coreos.com/jenkins-jobs-ref: ${JENKINS_JOBS_REF}
```

Comment:
we added these two annotations as well?

Reply:
Yeah sorry, I should've made that a separate commit. I'm not a fan myself of sneaking things into blocks that initially just looks like it was a plain move. :)

I added a comment there. Let me know if it's not clear!